src/components/Shopping/ShoppingCart.jsx

```diff
@@ -1,74 +1,68 @@
-import React, { useState } from "react";
+import React, { useEffect } from "react";
 import data from "../../../data/products.json";
 import styled from "styled-components";
-import Image from "next/dist/client/image";
 
-const euroFormatter = new Intl.NumberFormat("de-de", {
+export const euroFormatter = new Intl.NumberFormat("de-de", {
   style: "currency",
   currency: "EUR",
 });
 
 export const ShoppingCart = (props) => {
-  return props.shoppingCart.map(({ id, variant }) => {
+  const totalPrice = props.shoppingCart.reduce((sumPrice, { id, variant }) => {
     const product = data.find((product) => id === product.id);
-    return (
-      <StyledCart key={id}>
-        <h4>{product.name}</h4>
-        <p>{product.description}</p>
-        <p>
-          {euroFormatter.format(product.variants[variant].price)} -{" "}
-          {product.variants[variant].duration} Tage
-        </p>
+    return sumPrice + product.variants[variant].price;
+  }, 0);
+
+  useEffect(() => {
+    props.setTotal(euroFormatter.format(totalPrice));
+  }, [totalPrice]);
```

Comment:
you could move the logic calculating the total price into `Offer` that would get rid of the need to even use a state to store it in the first place. Calculate the value there and pass it down to all components that need it.